Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md

```diff
@@ -0,0 +1,168 @@
+#Coding Standards and Guidelines
+
+
+##Naming Conventions
+
+####General Naming Conventions
+
+* Capitalize the first letter of each word in the identifier
+
+* Do use PascalCasing for all public member, type, and namespace names consisting of multiple words
+
+```cs
+
+    //Namespace
+    Amdocs.Ginger.Common.Repository;
+
+    //Type
+    public class StreamReader{...}
+
+    //Method
+    public void StartDriver(){...};
+```
+* Do chose readable identifier names e.g. Horizontal alignment is more English readable than alignment horizontal
+ 
+* Do favor readability over brevity e.g. CanScrollHorizontally is better than ScollableX
+
+* Avoid using identifiers that conflict with keywords of widely used programming languages
+
+* Do not use underscores, hyphens or any other non-alphanumeric characters to differentiate words
+
+```cs
+	// Correct	
+	public DateTime ClientAppointment;
+	public TimeSpan TimeLeft;
+	 
+	// Avoid
+	public DateTime client_Appointment;
+	public TimeSpan time_Left;
+```
+* Do not use Hungarian notation or any other type identification in identifiers.
+```cs
+e.g. 
+//Correct 
+int counter; 
+//Incorrect
+int iCounter;
+```
+
+
+* Do not use abbreviation or contractions as part of identifier names 
+e.g. Use GetWindow instead of GetWin
+
+
+#### Naming Classes and Interfaces
+
+* Use Noun or noun phrases for naming classes e.g. Employee 
+* Do not prefix class names with prefix “C”
+* Do prefix interface names with the letter I, to indicate that the type is an interface.
+* Name Source files according to their main classes
+
+
+
+####Variables naming 
+* Use pascal casing with “m” prefix for global variables
+```cs
+    private RunsetOperations mRunserOperations {get; set;};
+```
+
+* Use pascal casing with “c” prefix for constants
+
+```cs
+   const int cNULLStringLen = -1;
+```   
+
+* Use camel casing for method parameters and local variables
+
+```cs
+        public int AddTwoNumbers(int firstNumber, int secondNumber)
+        {
+            int outputValue = firstNumber - secondNumber;
+            return outputValue;
+        }
+```   
+
+* Do not explicitly specify a type of an enum or values of enums
+
+```cs
+	// Correct
+	public enum Direction
+	{
+	    North,
+	    East,
+	    South,
+	    West
+	}
+
+    // Don't
+	public enum Direction : long
+	{
+	    North = 1,
+	    East = 2,
+	    South = 3,
+	    West = 4
+	}
+```
+
+* Do not suffix enum names with Enum
+
+
+```cs	 
+	// Correct
+	public enum Coin
+	{
+	    Penny,
+	    Nickel,
+	    Dime,
+	    Quarter,
+	    Dollar
+	} 
+ 
+	// Don't
+     public enum CoinEnum
+	{
+	    Penny,
+	    Nickel,
+	    Dime,
+	    Quarter,
+	    Dollar
+	}
+```
+
+####XAML Naming conventions
+* Use “x” prefix all XAML control names
+* Use use the type name as suffix for a xaml control names e.g. Btn for Button. 
+ So name of reset button in xaml will be "xResetBtn" 
+
+#### Ginger namespace
```

Comment:
NameSpaces will be in another doc - we can have global ref here to say check NameSpcaes

Reply:
Added a link reference to the namespace document